benchmarks/python/core.py

```diff
@@ -157,22 +160,19 @@ def _get_kernel_time(
         Returns:
             time_value: Elapsed CUDA time in seconds.
         """
-        elapsed_cuda_time = (
-            sum(
-                [
-                    # Re: torch profiler API changes in https://github.com/pytorch/pytorch/pull/123247
-                    (
-                        event.self_device_time_total
-                        if hasattr(event, "self_device_time_total")
-                        else event.self_cuda_time_total
-                    )
-                    for event in prof_averages
-                    if event.device_type == DeviceType.CUDA
-                ]
+        elapsed_cuda_time = 0
+        self.has_cuda_event = False
+        for event in prof_averages:
+            if event.device_type != DeviceType.CUDA:
+                continue
+            has_cuda_event = True
+            # Re: torch profiler API changes in https://github.com/pytorch/pytorch/pull/123247
+            elapsed_cuda_time = (
+                elapsed_cuda_time + event.self_device_time_total
+                if hasattr(event, "self_device_time_total")
+                else event.self_cuda_time_total
```

Comment:
We are currently seeing tests with no CUDA events -- which should not be happening. Rather than returning 0, we may want to add an assert here `NVF_CHECK(has_cuda_event)` to identify these unexpected behaviors.

Reply:
Yeah, I will add that assert back. I had removed the assert here because I was trying to get automatic calibration working. Since I never got that to work properly anyway and actually reflect the CUPTI precision properly, I think it's fine to just assert that if we're using this timer then there should always be some CUDA events between observations.